latex.bzl

```diff
@@ -79,8 +79,77 @@ _latex = rule(
     implementation = _latex_impl,
 )
 
-def latex_document(name, main, srcs = [], tags = [], cmd_flags = [], format="pdf"):
+def _latex_to_svg_impl(ctx):
+    toolchain = ctx.toolchains["@bazel_latex//:latex_toolchain_type"].latexinfo
+   
+    custom_dependencies = []
+    for deps in ctx.attr.deps:
+        for file in deps.files.to_list():
+            if file.dirname not in custom_dependencies:
+                custom_dependencies.append(file.dirname)
+    custom_dependencies = ','.join(custom_dependencies)
+
+    src = ctx.attr.src
+    if LatexOutputInfo in src:
+        input_file = src[LatexOutputInfo].file
+        input_format = src[LatexOutputInfo].format
+    else:
+        fail("LatexOutputInfo provider not available in src")
+
+    flags = []
+    if "pdf" in input_format:
+        flags.append("--flag=--pdf")
+    for cmd in ctx.attr.cmd_flags:
+       flags.append("--flag="+cmd) 
+ 
+    ctx.actions.run(
+        mnemonic = "DviSvgM",
+        use_default_shell_env = True,
+        executable = ctx.executable._tool,
+        arguments = [
+            "--dep-tool=" + toolchain.kpsewhich.files.to_list()[0].path,
+            "--tool=" + toolchain.dvisvgm.files.to_list()[0].path,
+            "--env=LIBGS" + ":" + ctx.files._libgs[0].dirname + "/lib/libgs.so",
+            "--input=" + input_file.path,
+            "--output=" + ctx.outputs.out.path,
+            "--tool-output=" + input_file.basename.rsplit(".", 1)[0] + ".svg",
+            "--inputs=" + custom_dependencies,
+        ] + flags,
+        inputs = depset(
+            direct = ctx.files.src + 
+                     ctx.files.deps +
+                     [toolchain.dvisvgm.files.to_list()[0]] +
+                     ctx.files._libgs,
+            transitive = [
+                toolchain.kpsewhich.files,
+                toolchain.dvisvgm.files,
+            ],
+        ),
+        outputs = [ctx.outputs.out],
+        tools = [ctx.executable._tool],
+    )
 
+_latex_to_svg = rule(
+    attrs = {
+        "src": attr.label(),
+        "deps": attr.label_list(allow_files = True),
+        "cmd_flags": attr.string_list(
```

Comment:
I would prefer not reproducing this pattern here - if there's specific behaviours we'd like to expose, I think it would be better exposing them through other specific arguments rather than allowing an arbitrary list of flags.

Reply:
Ack, I’ll remove it.